build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -46,12 +46,14 @@ public class CompileTargets
             { "osx.10.11-x64", "osx.10.10-x64" },
             { "ubuntu.14.04-x64", "ubuntu.14.04-x64" },
             { "ubuntu.16.04-x64", "ubuntu.16.04-x64" },
+            { "ubuntu.16.10-x64", "ubuntu.16.10-x64" },
             { "centos.7-x64", "rhel.7-x64" },
             { "rhel.7-x64", "rhel.7-x64" },
             { "rhel.7.2-x64", "rhel.7-x64" },
             { "debian.8-x64", "debian.8-x64" },
             { "fedora.23-x64", "fedora.23-x64" },
-            { "opensuse.13.2-x64", "opensuse.13.2-x64" }
+            { "opensuse.13.2-x64", "opensuse.13.2-x64" },
+            { "opensuse.42.1-x64", "opensuse.42.1-x64" }
```

Comment:
I was planning on adding the new distros in a separate change, because I still need to go through lots of test files, add the dockerfiles, etc.

Will this change cause issues if that stuff isn't already done? If so, I can revert this part and include it in the follow-up.